core/src/com/g16/feyrune/map/MapParser.java

```diff
@@ -0,0 +1,200 @@
+package com.g16.feyrune.map;
+
+import com.sun.tools.javac.util.Pair;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import javax.xml.XMLConstants;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import java.io.File;
+import java.util.ArrayList;
+
+/**
+ * This class handle all operations related to the parsing of map files.
+ */
+public class MapParser {
+    /**
+     * This method parses a map filed (produced by Tiled).
+     *
+     * @param filePath The file path of the map you want to parse.
+     * @return A {@link Map} object containing the parsed map.
+     */
+    public static Map parseMapFile(String filePath) {
+        Document doc = readXMLDocument(filePath);
+
+        Pair<Integer, Integer> mapSize = getMapSize(doc);
+
+        ArrayList<Integer> collisionIds = parseCollisionIds(doc);
+
+        boolean[] collisionList = generateCollisionList(doc, collisionIds, mapSize);
+
+        boolean[][] collisionMap = createCollisionMap(collisionList, mapSize);
+
+        return generateTileMap(collisionMap);
+    }
+
+    /**
+     * This method generates a Tile map from a collision map.
+     *
+     * @param collisionMap The collision map.
+     * @return A {@link Map} based on the collision map.
+     */
+    private static Map generateTileMap(boolean[][] collisionMap) {
```

Comment:
Does this really return ALL tiles used for rendering the map or does this only get tiles for the ones that have collition? 
I think that there is a point to have this or another function return ALL tiles. As i understand this, if we were to use this class to render a map we would not be able to see any tiles. 
Have i missunderstood the function of this class or are we missing something. 
I belive that the ids have a correlation to where they are placed in the tilemap and it is this parses responsibility to decode this?

Reply:
Well, I'm in a bit of a pickle on this one. First I wanted it do *everything* when it comes to parsing of the map, including graphics. But since the project *need* to adhere to the MVC pattern, one could argue we would need a separate V class for parsing the actual graphics, while this one simply provides the map model.

Reply:
Though, technically, we could add an array of IDs for every single tile, thus only needing the View layer to figure out what ID correlates to which texture.

Reply:
Yes! The tile that is sent to the map shold have its id aswell!

Reply:
There can also be multiple layers so i think Map shold have a Layer[] and a Layer has Tile[][]

Reply:
This shold probably be the simplest and still adhere to the convention that we are aiming for.